### PR TITLE
NGEN Setup.Next

### PR DESCRIPTION
**Pull request by jmarolf**

### CODE REVIEW

**Comment by jmarolf** [comment]
@dotnet/roslyn-ide @brettfo @heejaechang

**Comment by heejaechang** [comment]
👍

**Comment by heejaechang** [comment]
retest windows_release_unit32_prtest please

**Comment by heejaechang** [comment]
I believe we need to add this path to probing path of vsn.exe ?

**Comment by jmarolf** [comment]
@heejaechang setting ShortcutPath in the vsix tells the VS installer to add that folder to the list of known probing paths for ngen.

**Comment by heejaechang** [comment]
@jmarolf I see. but it looks like order of things appear in the probing path matter. how can one specify ordering?

File: src/VisualStudio/Setup.Next/source.extension.vsixmanifest [hunk 1]
```diff
@@ -4,13 +4,21 @@
...
   </Metadata>
   <Installation Experimental="true">
     <InstallationTarget Version="[15.0,]" Id="Microsoft.VisualStudio.Pro" />
     <InstallationTarget Version="[15.0,]" Id="Microsoft.VisualStudio.VSWinDesktopExpress" />
     <InstallationTarget Version="[15.0,]" Id="Microsoft.VisualStudio.VWDExpress" />
     <InstallationTarget Version="[15.0,]" Id="Microsoft.VisualStudio.VSWinExpress" />
   </Installation>
+  <Installer>
+    <Actions>
+      <Action Type="Ngen" Path="Microsoft.CodeAnalysis.EditorFeatures.Next.dll" />
```
Comment (heejaechang) [inline src/VisualStudio/Setup.Next/source.extension.vsixmanifest]: looks like uses |EditorFeatures.Next| thing to find out actual name, but require hard coding name of dll. will it be a problem?

**Comment by jmarolf** [comment]
@jasonmalinowski everything installs to a path and has a list of assemblies to ngen except for the new out-of-proc manifest.

File: src/VisualStudio/Setup.Next/source.extension.vsixmanifest (hunk 1 above)
Comment (jmarolf) [inline src/VisualStudio/Setup.Next/source.extension.vsixmanifest]: the ngen action is given a path, it cannot look it up and replace the path with a reference to a given project.  We should file a bug on V3 vsix tooling if we want this.

**Pull request merged**